### ISSUE TITLE
Router: Navigation ends before components in route is created

### ISSUE THREAD

**Issue by andershessellund** [angular/angular#49996]
### Which @angular/* package(s) are the source of the bug?

router

### Is this a regression?

Yes

### Description

I bootstrap an angular app using the new bootstrapApplication(...) method. The bootstrapped component only contains a router-outlet. I lazy-load a module into this router outlet. The lazy loaded module has two components, one is a child of the other.

When I navigate to this child component, the navigation is "completed" before the child component is constructed, that is, NavigationEnd is emitted before the child component is constructed, and within the child component constructor I cannot access Router.currentNavigation().

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/angular-nkuol7?file=src/main.ts

### Please provide the exception or error you saw

```true
In the reproduction of the bug, I would expect to see "constructing module-child" in the console log *before* the NavigationEnd event is logged.
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 15.2.6
Node: 18.13.0
Package Manager: npm 8.19.3
OS: darwin arm64

Angular: 15.2.8
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1502.6
@angular-devkit/build-angular   15.2.6
@angular-devkit/core            15.2.6
@angular-devkit/schematics      15.2.6
@angular/cli                    15.2.6
@schematics/angular             15.2.6
rxjs                            7.8.0
typescript                      4.9.5
```

**Comment by atscott** [comment]
That’s correct, this is working as expected and listed in the breaking changes for v15 https://angular.io/guide/update-to-version-15#routeroutlet-instantiates-the-component-after-change-detection
